test/ActionRequestTest.ts

```diff
@@ -1,131 +1,242 @@
+    
+                const originalRequestSent = httpBinResponse.json;
+    
+                assert.equal(originalRequestSent.result.resolvedQuery, "actions_intent_PERMISSION");
+                assert.equal(originalRequestSent.lang, "en-UK");
+    
+            });
+    
+            it("Reset the filters action", async () => {
+                const virtualGoogle = VirtualGoogleAssistant.Builder()
+                    .actionUrl( "https://httpbin.org/post" )
+                    .directory("./test/resources/sampleProject")
+                    .create();
+    
+                virtualGoogle.addFilter((request) => {
+                    request.lang = "en-UK";
+                });
+    
+                virtualGoogle.resetFilters();
+    
+                const httpBinResponse = await virtualGoogle.utter("what is the pokemon at 25");
+    
+                const originalRequestSent = httpBinResponse.json;
+    
+                assert.equal(originalRequestSent.lang, "en-us");
+            });
+    
+            it("Sent the request in other locale", async () => {
+                const virtualGoogle = VirtualGoogleAssistant.Builder()
+                    .actionUrl( "https://httpbin.org/post" )
+                    .directory("./test/resources/sampleProject")
+                    .locale("en-UK")
+                    .create();
+    
+                const httpBinResponse = await virtualGoogle.utter("what is the pokemon at 25");
+    
+                const originalRequestSent = httpBinResponse.json;
+    
+                assert.equal(originalRequestSent.lang, "en-UK");
             });
-
-            virtualGoogle.resetFilters();
-
-            const httpBinResponse = await virtualGoogle.utter("what is the pokemon at 25");
-
-            const originalRequestSent = httpBinResponse.json;
-
-            assert.equal(originalRequestSent.lang, "en-us");
         });
 
-        it("Sent the request in other locale", async () => {
-            const virtualGoogle = VirtualGoogleAssistant.Builder()
-                .actionUrl( "https://httpbin.org/post" )
-                .directory("./test/resources/sampleProject")
-                .locale("en-UK")
-                .create();
-
-            const httpBinResponse = await virtualGoogle.utter("what is the pokemon at 25");
-
-            const originalRequestSent = httpBinResponse.json;
-
-            assert.equal(originalRequestSent.lang, "en-UK");
+        describe("DialogFlow v2", () => {
+            const model: InteractionModel = InteractionModel.fromDirectory("./test/resources/sampleProjectv2");
+            const requestGenerator: ActionRequest = new ActionRequestV2(model, "en-us");
+
+            it("For a intent with slots", () => {
+                const request = requestGenerator.intentRequest("PokedexIntent").withSlot("number", "22");
+                // assert.equal(request.toJSON().queryResult.intent.number, 22);
```

Comment:
remove the commented code, but also this is missing for the check to see if 22 is on the correct attribute